backend/api/v1/bookmark_service.go

```diff
@@ -0,0 +1,112 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/bytebase/bytebase/backend/common"
+
+	"github.com/bytebase/bytebase/backend/store"
+	v1pb "github.com/bytebase/bytebase/proto/generated-go/v1"
+)
+
+// BookmarkService implments the v1pb.BookmarkServiceServer interface.
+type BookmarkService struct {
+	v1pb.UnimplementedBookmarkServiceServer
+	store *store.Store
+}
+
+// NewBookmarkService returns a new BookmarkService.
+func NewBookmarkService(store *store.Store) *BookmarkService {
+	return &BookmarkService{
+		store: store,
+	}
+}
+
+// CreateBookmark creates a new bookmark.
+func (s *BookmarkService) CreateBookmark(ctx context.Context, request *v1pb.CreateBookmarkRequest) (*v1pb.Bookmark, error) {
+	currentPincipalUID := ctx.Value(common.PrincipalIDContextKey).(int)
+	princialUID, err := getUserID(request.Parent)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid parent: %v", err)
+	}
+	if princialUID != currentPincipalUID {
+		return nil, status.Errorf(codes.PermissionDenied, "Permission denied")
+	}
+	if request.Bookmark == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Missing bookmark")
```

Comment:
not found?

Reply:
Using Invalid Argument because we are in the check argument stage. We should return not found if we cannot find the book mark in store.